### PR TITLE
RWA-1450 refactor perform operations logic to make it extensible

### PR DESCRIPTION
**Pull request by vgulapally**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1450

### Change description ###
refactor perform operations logic to make it extensible

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by rasikasri — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskOperationRequest(TaskOperationName.EXECUTE_RECONFIGURE, OffsetDateTime.now().minus(Duration.ofDays(1))),
+            caseworkerCredentials.getHeaders()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:74]: can we also include checks on the verifiable task attributes that confirm the operation has completed successfully and as expected? I.e 
task state = assigned
reconfigure_request_time = null
last_reconfiguration_time = now()

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/ExecuteReconfigureTasksControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        when(configureTaskService.configureCFTTask(any(TaskResource.class), any(TaskToConfigure.class)))
+            .thenReturn(testTaskResource);
+
+        when(taskAutoAssignmentService.reAutoAssignCFTTask(any(TaskResource.class)))
+            .thenReturn(testTaskResource);
+
+    }
+
+    @Test
+    void should_execute_reconfigure_for_reconfigure_request_time() throws Exception {
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/ExecuteReconfigureTasksControllerTest.java:112]: As this is doing both operations, should we make it clear via the test name? should_successfully_mark_to_configure_and_then_execute_reconfigure_on_task

Comment (vgulapally) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/ExecuteReconfigureTasksControllerTest.java]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java (hunk 1 above)
Comment (vgulapally) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:81]: done

**Review by dharmendrak — DISMISSED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            "task/{task-id}/assign",
+            taskVariables.getTaskId(),
+            new AssignTaskRequest(assigneeId),
+            assignerCredentials.getHeaders()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        switch (testGrantType) {
```
Comment (serdaroguzhanakin) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:220]: is this necessary?

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            "task/{task-id}/assign",
+            taskVariables.getTaskId(),
+            new AssignTaskRequest(assigneeId),
+            assignerCredentials.getHeaders()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        switch (testGrantType) {
```
[the remark this answers is not in the document]
Comment (Lee365) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:171]: Corrected this to remove switch.

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java (hunk 3 above)
Comment (Lee365) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:220]: Corrected this to remove switch.

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+
+        common.setupHearingPanelJudgeForSpecificAccess(assignerCredentials.getHeaders(),
+                                                       taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
+        initiateTask(assignerCredentials.getHeaders(), taskVariables,
+                     "processApplication", "process application", "process task");
+
+        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(),
+                                                 WA_JURISDICTION, WA_CASE_TYPE);
+        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
```
[the remark this answers is not in the document]
Comment (Lee365) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:78]: TribunalCaseworker role can mark task to reconfigure.

**Review by dharmendrak — DISMISSED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+        result.prettyPrint();
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+            .and().body("task.id", equalTo(taskId))
+            .body("task.task_state", is("assigned"))
+            .body("task.reconfigure_request_time", notNullValue());
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:105]: .body("task.last_reconfiguration_time", nullValue()); check?

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+        result.prettyPrint();
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+            .and().body("task.id", equalTo(taskId))
+            .body("task.task_state", is("assigned"))
+            .body("task.reconfigure_request_time", notNullValue());
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:89]: .body("task.last_reconfiguration_time", nullValue());?  check

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        result.prettyPrint();
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+            .and().body("task.id", equalTo(taskId))
+            .body("task.task_state", is("unassigned"))
+            .body("task.reconfigure_request_time", notNullValue());
+
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:127]: .body("task.last_reconfiguration_time", nullValue());? check

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java (hunk 6 above)
Comment (vgulapally) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskExecuteReconfigureControllerCFTTest.java:106]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java (hunk 7 above)
Comment (vgulapally) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:91]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java (hunk 8 above)
Comment (vgulapally) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskMarkReconfigureControllerCFTTest.java:130]: done

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**